src/Microsoft.AspNetCore.ResponseCaching/ResponseCachingContext.cs

```diff
@@ -227,9 +249,25 @@ internal void FinalizeCachingBody()
         {
             if (State.ShouldCacheResponse && ResponseCacheStream.BufferingEnabled)
             {
-                State.CachedResponse.Body = ResponseCacheStream.BufferedStream.ToArray();
+                if (ResponseCacheStream.BufferedStream.Length >= _options.MinimumSplitBodySize)
```

Comment:
Offline discussion, we decided that the accuracy here is not significant. Header sizes are relatively small and quite consistent.